### ISSUE TITLE
pygrib broken with numpy v2

### ISSUE THREAD

**Issue by rafa-guedes** [jswhit/pygrib#251]
pygrib is broken in my environment with numpy v2. I have noticed this binary incompatibility before as a warning instead of ValueError so perhaps this became less tolerable in the new numpy release?

```
In [1]: import pygrib
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[1], line 1
----> 1 import pygrib

File ~/.virtualenvs/tmp/lib/python3.10/site-packages/pygrib/__init__.py:2
      1 # init for pygrib package
----> 2 from ._pygrib import *
      3 # Need explicit imports for names beginning with underscores
      4 from ._pygrib import __doc__, __version__

File src/pygrib/_pygrib.pyx:1, in init pygrib._pygrib()

ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
```

**Comment by jerryfrancis-97** [comment]
Is there any update on this?

**Comment by rafa-guedes** [comment]
https://github.com/numpy/numpy/issues/26710

**Comment by leaver2000** [comment]
It appears this can be fixed by changing `oldest-supported-numpy` ➡️ `numpy` in the `pyproject.toml`. I am unfamiliar with `oldest-supported-numpy` so I don't know how this might behave on other systems.

# current pyproject.toml

```toml
[build-system]
requires = [
    "Cython>=0.29",
    "oldest-supported-numpy",
    "setuptools>=61",
]
```

# 🛑 catastrophic failure

```bash
.venv λ › pip install . --upgrade && cd test/ && python run_tests.py && cd ..
Processing /home/leaver/pygrib
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: packaging in ./.venv/lib/python3.12/site-packages (from pygrib==2.1.6) (24.1)
Requirement already satisfied: pyproj in ./.venv/lib/python3.12/site-packages (from pygrib==2.1.6) (3.6.1)
Requirement already satisfied: numpy in ./.venv/lib/python3.12/site-packages (from pygrib==2.1.6) (2.0.0)
Requirement already satisfied: certifi in ./.venv/lib/python3.12/site-packages (from pyproj->pygrib==2.1.6) (2024.6.2)
Building wheels for collected packages: pygrib
  Building wheel for pygrib (pyproject.toml) ... done
  Created wheel for pygrib: filename=pygrib-2.1.6-cp312-cp312-linux_x86_64.whl size=1631697 sha256=fcd2f0571fba1c3b14f0d1747dec0504670b693a03f8e440ee3c08d15ca08188
  Stored in directory: /tmp/pip-ephem-wheel-cache-_iydvv4s/wheels/1b/e8/82/1b48e18dba8406ec2ce7b710860a7c31ed844f5f00c65bc5e1
Successfully built pygrib
Installing collected packages: pygrib
  Attempting uninstall: pygrib
    Found existing installation: pygrib 2.1.6
    Uninstalling pygrib-2.1.6:
      Successfully uninstalled pygrib-2.1.6
Successfully installed pygrib-2.1.6
running test_stere.py...
Traceback (most recent call last):
  File "/home/leaver/pygrib/test/run_tests.py", line 9, in <module>
    exec(open(f).read())
  File "<string>", line 1, in <module>
  File "/home/leaver/pygrib/.venv/lib/python3.12/site-packages/pygrib/__init__.py", line 2, in <module>
    from ._pygrib import *
  File "src/pygrib/_pygrib.pyx", line 1, in init pygrib._pygrib
ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
```

# one line change

```toml
[build-system]
requires = [
    "Cython>=0.29",
    "numpy",
    "setuptools>=61",
]
```

# ✅ great success

```bash
.venv λ › pip install . --upgrade && cd test/ && python run_tests.py && cd ..
Processing /home/leaver/pygrib
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: packaging in ./.venv/lib/python3.12/site-packages (from pygrib==2.1.6) (24.1)
Requirement already satisfied: pyproj in ./.venv/lib/python3.12/site-packages (from pygrib==2.1.6) (3.6.1)
Requirement already satisfied: numpy in ./.venv/lib/python3.12/site-packages (from pygrib==2.1.6) (2.0.0)
Requirement already satisfied: certifi in ./.venv/lib/python3.12/site-packages (from pyproj->pygrib==2.1.6) (2024.6.2)
Building wheels for collected packages: pygrib
  Building wheel for pygrib (pyproject.toml) ... done
  Created wheel for pygrib: filename=pygrib-2.1.6-cp312-cp312-linux_x86_64.whl size=1630728 sha256=c4e90e791873a2fba58b47209e8300929e53c44a07b1d3e200dd33151963dbd4
  Stored in directory: /tmp/pip-ephem-wheel-cache-bexokrl3/wheels/1b/e8/82/1b48e18dba8406ec2ce7b710860a7c31ed844f5f00c65bc5e1
Successfully built pygrib
Installing collected packages: pygrib
  Attempting uninstall: pygrib
    Found existing installation: pygrib 2.1.6
    Uninstalling pygrib-2.1.6:
      Successfully uninstalled pygrib-2.1.6
Successfully installed pygrib-2.1.6
running test_stere.py...
running test_reglatlon.py...
running test_ndfd_pr.py...
running test_ndfd_conus.py...
running test_set_bitmap.py...
running test_spectral.py...
skipping test that requires pyspharm (python spherical harmonic module) from http://code.google.com/p/pyspharm
```

for reference...

```
.venv λ › python -V && pip freeze 
Python 3.12.3
Cartopy==0.23.0
certifi==2024.6.2
contourpy==1.2.1
cycler==0.12.1
Cython==3.0.10
fonttools==4.53.0
iniconfig==2.0.0
kiwisolver==1.4.5
matplotlib==3.9.0
numpy==2.0.0
packaging==24.1
pillow==10.3.0
pluggy==1.5.0
pygrib @ file:///home/leaver/pygrib
pyparsing==3.1.2
pyproj==3.6.1
pyshp==2.3.1
pytest==8.2.2
python-dateutil==2.9.0.post0
setuptools==70.1.0
shapely==2.0.4
six==1.16.0
wheel==0.43.0
```

```bash
.venv λ › cat /etc/os-release         
PRETTY_NAME="Ubuntu 22.04.4 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.4 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

**Comment by leaver2000** [comment]
Upon further investigation it appears there is a [deprecation-notice-for-numpy-20](https://github.com/scipy/oldest-supported-numpy/tree/main?tab=readme-ov-file#deprecation-notice-for-numpy-20). I'll open a PR with this change.

> Until NumPy 1.25.0 this package was needed to be able to target the oldest supported NumPy version for packages using the NumPy C API. That targeting was necessary in order to obtain wheel for distribution on PyPI. However, a new "C API exporting" feature in NumPy 1.25.0 made that unnecessary. And with NumPy 2.0, it is now actually necessary to use >=2.0 at build time for packages that want to support both NumPy 1.xx and 2.0.

> All users should migrate to expressing a build dependency on numpy directly, according to the guidance given in

**Comment by jswhit** [comment]
fixed in PR #254

**Comment by nebukadnezar** [comment]
when is that fix going to be on pypi? Just installed from there an hour ago and ran into this same problem posted above.

**Comment by nebukadnezar** [comment]
Ok for anyone else equally manual installation unsavvy as myself, here's how I solved it for installation into my venv:

```
python3 -m venv myvenv
source myvenv/bin/activate
pip install eccodes
pip install setuptools
pip install cython
git clone https://github.com/jswhit/pygrib
cd pygrib
python setup install
```

and it seems to work from the venv

**Comment by jswhit** [comment]
Closed by PR #254.   It will take me some time to update wheels on pypi.

**Comment by jswhit** [comment]
> when is that fix going to be on pypi? Just installed from there an hour ago and ran into this same problem posted above.

new wheels on pypi now
